### ISSUE TITLE
Resolve UNLs with short paths relative to c

### ISSUE THREAD

**Issue by edreamleo** [leo-editor/leo-editor#3814]
See PR #3815. In the discussion of PR #3812, @boltex said:

[My] intended enhancement was that short UNLs could find and *open* outlines if the short UNL could be resolved relative to the invoking commander's current path, without needing an entry in `@data unl-path-prefixes`.

**Discussion (EKR)**

On second reading, I agree that this proposal makes sense.  It involves a change only to `g.openUNLFile`.

**My mistaken first impression**

My first (erroneous) thought was that this issue suggested that Leo should support **relative paths** in UNLs.
These paths would be resolved using the directory containing `c`, the presently selected outline.

This would be a significant change to Leo's UNLs. It would affect Leo's gui and Leo's UNLs.
Imo, this proposal is not worth doing for the following reasons:

At present, legacy (path-based) and new (gnx-based) UNLs can have the following kinds of file parts:

- Empty. Resolves to `c`, the presently selected outline.
- Short. Resolves using `c`, the outlines listed in `@data unl-path-prefixes`, and all other open outlines.
- Long (absolute). Resolves *only* the the given outline.

Imo, this existing behaviour is adequate to cover all reasonable use cases.
In particular, sessions are an easy way to open any outlines needed to resolve UNLs with short file parts.

This proposal's **fatal flaw** it that creates a *less convenient* work flow for users.

Indeed, Leo *never* generates relative file parts because *all UNLs in Leo's status area refer to c*.
In other words, users would have to create relative file parts **by hand**.

Yes, we could imagine new (complex, never used) machinery for specifying relative file parts, but that would be pure featuritis.

**Comment by boltex** [comment]
@edreamleo @tbpassin **Thank you dear friends! IT WORKS!** 🥳 🎺 🎵 Hurray!!

Now, I'm off to transliterate all that UNL good stuff to LeoJS!!  (And finish my second video tutorial, where the opening scene is me using a UNL in a 'table of contents Leo document' to open another Leo file that has some demonstrations of using the VSCode UI from within leo scripts! - point being to obviously also demonstrating the existence and usage of UNLs among all the other features I present in the video! )

@edreamleo In a month or so, When we're done with Leo 6.7.8, and LeoJS 1.0, and the dust has settled... I'd like to revisit the reasons you gave against general "relative-resolution" of paths for short UNL because I disagree with most of the reasons that you laid out, except one: _users would have to create relative file parts by hand for crafting such UNLs in those specific instances._  which to me, is true, but not significant compared to the trouble of the receiving part of a user-to-user project sharing, which would contain UNLs. 

Anyways, **no rush,** that other aspect can wait, as the basic functionality of a Leo document being able to 'see and open' another one laying just beside it now works ! 😄 **Many thanks again!!!**
